### PR TITLE
Native probe support for AT-Proto

### PR DESCRIPTION
**Pull request by annando**
Since the AT-Protocol handling is now in the core, we can move the probing to the core as well.

### CODE REVIEW

File: src/DI.php [hunk 1]
```diff
@@ -157,7 +157,7 @@ public static function atProtocol()
 	}
 
 	/**
-	 * @return AtProtocol\Arguments
+	 * @return AtProtocol\Actor
```
Comment (MrPetovan) [inline src/DI.php]:
```suggestion
	 * @return ATProtocol\Actor
```

File: src/Protocol/ATProtocol.php [hunk 2]
```diff
@@ -66,6 +66,11 @@ public function __construct(LoggerInterface $logger, Database $database, IManage
 		$this->httpClient = $httpClient;
 	}
 
+	/**
+	 * Returns an array of user ids who want to import the Bluesky timeline
+	 *
+	 * @return array
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]: No need to document parameters and return values that are already typed in the code if there isn't an associated comment with them.
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 3]
```diff
@@ -92,6 +97,15 @@ public function getUids(): array
...
 	}
 
+	/**
+	 * Fetches XRPC data
+	 * @see https://atproto.com/specs/xrpc#lexicon-http-endpoints
+	 *
+	 * @param string $url
+	 * @param array $parameters
+	 * @param integer $uid
+	 * @return stdClass|null
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 4]
```diff
@@ -119,6 +133,13 @@ public function XRPCGet(string $url, array $parameters = [], int $uid = 0): ?std
 		return $data;
 	}
 
+	/**
+	 * Fetch data from the given URL via GET and return it as a JSON class
+	 *
+	 * @param string $url
+	 * @param array $opts
+	 * @return stdClass|null
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 5]
```diff
@@ -141,12 +162,30 @@ public function get(string $url, array $opts = []): ?stdClass
...
+	 * @param integer $uid
+	 * @param string $url
+	 * @param [type] $parameters
+	 * @return stdClass|null
+	 */
 	public function XRPCPost(int $uid, string $url, $parameters): ?stdClass
 	{
 		$data = $this->post($uid, '/xrpc/' . $url, json_encode($parameters), ['Content-type' => 'application/json', 'Authorization' => ['Bearer ' . $this->getUserToken($uid)]]);
 		return $data;
 	}
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]: This method doesn't appear to be used anywhere.
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 6]
```diff
@@ -141,12 +162,30 @@ public function get(string $url, array $opts = []): ?stdClass
...
 
+	/**
+	 * Post data to the user PDS
+	 *
+	 * @param integer $uid
+	 * @param string $url
+	 * @param string $params
+	 * @param array $headers
+	 * @return stdClass|null
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 7]
```diff
@@ -180,6 +219,13 @@ public function post(int $uid, string $url, string $params, array $headers): ?st
 		return $data;
 	}
 
+	/**
+	 * Fetches the PDS for a given user
+	 * @see https://atproto.com/guides/glossary#pds-personal-data-server
+	 *
+	 * @param integer $uid
+	 * @return string|null
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 8]
```diff
@@ -205,6 +251,14 @@ private function getUserPds(int $uid): ?string
 		return $pds;
 	}
 
+	/**
+	 * Fetch the DID for a given user
+	 * @see https://atproto.com/guides/glossary#did-decentralized-id
+	 *
+	 * @param integer $uid
+	 * @param boolean $refresh
+	 * @return string|null
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 9]
```diff
@@ -233,6 +287,12 @@ public function getUserDid(int $uid, bool $refresh = false): ?string
 		return $did;
 	}
 
+	/**
+	 * Fetches the DID for a given handle
+	 *
+	 * @param string $handle
+	 * @return string
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 10]
```diff
@@ -268,6 +328,12 @@ public function getDid(string $handle): string
 		return '';
 	}
 
+	/**
+	 * Fetches a DID for a given profile URL
+	 *
+	 * @param string $url
+	 * @return string
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 11]
```diff
@@ -317,6 +383,13 @@ public function getDidByProfile(string $url): string
 		return $ids['bsky_did'];
 	}
 
+	/**
+	 * Fetches the DID of a given handle via a HTTP request to the .well-known URL.
+	 * This is one of the ways, custom handles can be authorized.
+	 *
+	 * @param string $handle
+	 * @return string
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 12]
```diff
@@ -331,6 +404,13 @@ private function getDidByWellknown(string $handle): string
 		return '';
 	}
 
+	/**
+	 * Fetches the DID of a given handle via a DND request.
+	 * This is one of the ways, custom handles can be authorized.
+	 *
+	 * @param string $handle
+	 * @return string
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 13]
```diff
@@ -350,7 +430,13 @@ private function getDidByDns(string $handle): string
 		return '';
 	}
 
-	private function getPdsOfDid(string $did): ?string
+	/**
+	 * Fetch the PDS of a given DID
+	 *
+	 * @param string $did
+	 * @return string|null
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 14]
```diff
@@ -366,6 +452,13 @@ private function getPdsOfDid(string $did): ?string
 		return null;
 	}
 
+	/**
+	 * Checks if the provided DID matches the handle
+	 *
+	 * @param string $did
+	 * @param string $handle
+	 * @return boolean
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 15]
```diff
@@ -376,6 +469,12 @@ private function isValidDid(string $did, string $handle): bool
 		return in_array('at://' . $handle, $data->alsoKnownAs);
 	}
 
+	/**
+	 * Fetches the user token for a given user
+	 *
+	 * @param integer $uid
+	 * @return string
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 16]
```diff
@@ -390,6 +489,12 @@ public function getUserToken(int $uid): string
 		return $token;
 	}
 
+	/**
+	 * Refresh and returns the user token for a given user.
+	 *
+	 * @param integer $uid
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol.php [hunk 17]
```diff
@@ -412,6 +517,13 @@ private function refreshUserToken(int $uid): string
 		return $data->accessJwt;
 	}
 
+	/**
+	 * Create a user token for the given user
+	 *
+	 * @param integer $uid
+	 * @param string $password
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol.php]:
```suggestion
```

File: src/Protocol/ATProtocol/Actor.php [hunk 18]
```diff
@@ -35,6 +35,12 @@ public function __construct(LoggerInterface $logger, ATProtocol $atprotocol)
 		$this->atprotocol = $atprotocol;
 	}
 
+	/**
+	 * Syncronize the contacts (followers, sharers) for the given user
+	 *
+	 * @param integer $uid
+	 * @return void
+	 */
 	public function syncContacts(int $uid)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Actor.php]:
```suggestion
	 */
	public function syncContacts(int $uid): void
```

File: src/Protocol/ATProtocol/Actor.php [hunk 19]
```diff
@@ -93,9 +99,16 @@ public function syncContacts(int $uid)
...
 
-	public function updateContactByDID(string $did)
+	/**
+	 * Update a contact for a given DID and user id
+	 *
+	 * @param string $did
+	 * @param integer $contact_uid
+	 * @return void
+	 */
+	public function updateContactByDID(string $did, int $contact_uid)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Actor.php]:
```suggestion
	 */
	public function updateContactByDID(string $did, int $contact_uid): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 20]
```diff
@@ -74,9 +73,11 @@ public function __construct(LoggerInterface $logger, IManageConfigValues $config
...
 
-	// *****************************************
-	// * Listener
-	// *****************************************
+	/**
+	 * Listen to incoming webstream messages from Jetstream
+	 *
+	 * @return void
+	 */
 	public function listen()
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	public function listen(): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 21]
```diff
@@ -137,6 +138,11 @@ public function listen()
 		}
 	}
 
+	/**
+	 * Increment the message counter for the statistics page
+	 *
+	 * @return void
+	 */
 	private function incrementMessages()
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	private function incrementMessages(): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 22]
```diff
@@ -146,6 +152,11 @@ private function incrementMessages()
 		$this->keyValue->set('jetstream_messages', $packets + 1);
 	}
 
+	/**
+	 * Synchronize contacts for all active users
+	 *
+	 * @return void
+	 */
 	private function syncContacts()
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php:160]:
```suggestion
	 */
	private function syncContacts(): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 23]
```diff
@@ -158,6 +169,11 @@ private function syncContacts()
 		}
 	}
 
+	/**
+	 * Set options like the followed DIDs
+	 *
+	 * @return void
+	 */
 	private function setOptions()
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php:177]:
```suggestion
	 */
	private function setOptions(): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 24]
```diff
@@ -219,6 +235,15 @@ private function setOptions()
...
 	}
 
+	/**
+	 * Returns an array of DIDs provided by an array of contacts
+	 *
+	 * @param array $contacts
+	 * @param array $uids
+	 * @param integer $did_limit
+	 * @param array $dids
+	 * @return array
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 25]
```diff
@@ -233,6 +258,12 @@ private function addDids(array $contacts, array $uids, int $did_limit, array $di
 		return $dids;
 	}
 
+	/**
+	 * Route incoming messages
+	 *
+	 * @param stdClass $data
+	 * @return void
+	 */
 	private function route(stdClass $data)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	private function route(stdClass $data): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 26]
```diff
@@ -254,6 +285,12 @@ private function route(stdClass $data)
 		}
 	}
 
+	/**
+	 * Route incoming commit messages
+	 *
+	 * @param stdClass $data
+	 * @return void
+	 */
 	private function routeCommits(stdClass $data)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	private function routeCommits(stdClass $data): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 27]
```diff
@@ -304,6 +341,12 @@ private function routeCommits(stdClass $data)
 		}
 	}
 
+	/**
+	 * Calculate the drift between the server timestamp and the current time.
+	 *
+	 * @param stdClass $data
+	 * @return integer
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 28]
```diff
@@ -321,6 +364,13 @@ private function getDrift(stdClass $data): int
...
 	}
 
+	/**
+	 * Route app.bsky.feed.post commits 
+	 *
+	 * @param stdClass $data
+	 * @param integer $drift
+	 * @return void
+	 */
 	private function routePost(stdClass $data, int $drift)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	private function routePost(stdClass $data, int $drift): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 29]
```diff
@@ -338,6 +388,13 @@ private function routePost(stdClass $data, int $drift)
...
 	}
 
+	/**
+	 * Route app.bsky.feed.repost commits 
+	 *
+	 * @param stdClass $data
+	 * @param integer $drift
+	 * @return void
+	 */
 	private function routeRepost(stdClass $data, int $drift)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	private function routeRepost(stdClass $data, int $drift): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 30]
```diff
@@ -355,6 +412,12 @@ private function routeRepost(stdClass $data, int $drift)
 		}
 	}
 
+	/**
+	 * Route app.bsky.feed.like commits 
+	 *
+	 * @param stdClass $data
+	 * @return void
+	 */
 	private function routeLike(stdClass $data)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	private function routeLike(stdClass $data): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 31]
```diff
@@ -372,6 +435,12 @@ private function routeLike(stdClass $data)
 		}
 	}
 
+	/**
+	 * Route app.bsky.actor.profile commits 
+	 *
+	 * @param stdClass $data
+	 * @return void
+	 */
 	private function routeProfile(stdClass $data)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	private function routeProfile(stdClass $data): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 32]
```diff
@@ -393,6 +462,12 @@ private function routeProfile(stdClass $data)
 		}
 	}
 
+	/**
+	 * Route app.bsky.graph.follow commits 
+	 *
+	 * @param stdClass $data
+	 * @return void
+	 */
 	private function routeFollow(stdClass $data)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	private function routeFollow(stdClass $data): void
```

File: src/Protocol/ATProtocol/Jetstream.php [hunk 33]
```diff
@@ -416,6 +491,12 @@ private function routeFollow(stdClass $data)
 		}
 	}
 
+	/**
+	 * Store commit messages for debugging purposes
+	 *
+	 * @param stdClass $data
+	 * @return void
+	 */
 	private function storeCommitMessage(stdClass $data)
```
Comment (MrPetovan) [inline src/Protocol/ATProtocol/Jetstream.php]:
```suggestion
	 */
	private function storeCommitMessage(stdClass $data): void
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Protocol/ATProtocol.php (hunk 5 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: It is used in the addon.

File: src/DI.php (hunk 1 above)
Comment (annando) [reply src/DI.php]: Done

File: src/Protocol/ATProtocol.php (hunk 2 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added.

File: src/Protocol/ATProtocol.php (hunk 8 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 9 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 10 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 11 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 12 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 13 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 14 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 15 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 16 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

File: src/Protocol/ATProtocol.php (hunk 17 above)
Comment (annando) [reply src/Protocol/ATProtocol.php]: Description added

**Comment by annando** [comment]
@MrPetovan I added a lot of descriptions for the parameters.

**Comment by MrPetovan** [comment]
Thanks, in the future please don't amend your commits when you address feedback, it makes it harder for me to see what you actually changed.

**Review by MrPetovan — APPROVED** [review]

**Comment by annando** [comment]
Okay.

**Pull request merged**